BlueDB/src/main/java/org/bluedb/api/BlueCollection.java

```diff
@@ -9,25 +9,88 @@
 import org.bluedb.api.keys.BlueKey;
 import org.bluedb.api.keys.ValueKey;
 
+/**
+ * A BlueCollection is a persisted map of keys (of type BlueKey) to values of object type T.

```

Comment:
Consider `A BlueCollection represents a persisted map` or something like that. I want to make it obvious that the whole collection is not in memory, it just provides access to the collection that is persisted on disk.

We could also call it a persisted key/value datastore. But map is probably fine